can not build from source, when boost is boost_1_67_0 .

```
# cmake ..
-- The C compiler identification is GNU 4.8.5
-- The CXX compiler identification is GNU 4.8.5
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:744 (message):
  Imported targets not available for Boost version 106700
Call Stack (most recent call first):
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:848 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:1435 (_Boost_MISSING_DEPENDENCIES)
  cmake/EthDependencies.cmake:41 (find_package)
  CMakeLists.txt:22 (include)


CMake Warning at /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:744 (message):
  Imported targets not available for Boost version 106700
Call Stack (most recent call first):
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:848 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:1435 (_Boost_MISSING_DEPENDENCIES)
  cmake/EthDependencies.cmake:41 (find_package)
  CMakeLists.txt:22 (include)


CMake Warning at /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:744 (message):
  Imported targets not available for Boost version 106700
Call Stack (most recent call first):
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:848 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:1435 (_Boost_MISSING_DEPENDENCIES)
  cmake/EthDependencies.cmake:41 (find_package)
  CMakeLists.txt:22 (include)


CMake Warning at /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:744 (message):
  Imported targets not available for Boost version 106700
Call Stack (most recent call first):
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:848 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:1435 (_Boost_MISSING_DEPENDENCIES)
  cmake/EthDependencies.cmake:41 (find_package)
  CMakeLists.txt:22 (include)


CMake Warning at /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:744 (message):
  Imported targets not available for Boost version 106700
Call Stack (most recent call first):
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:848 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake-3.7/Modules/FindBoost.cmake:1435 (_Boost_MISSING_DEPENDENCIES)
  cmake/EthDependencies.cmake:41 (find_package)
  CMakeLists.txt:22 (include)


-- boost headers: /opt/boost/include
-- boost lib   : /opt/boost/lib/libboost_regex.a;/opt/boost/lib/libboost_filesystem.a;/opt/boost/lib/libboost_unit_test_framework.a;/opt/boost/lib/libboost_program_options.a;/opt/boost/lib/libboost_system.a
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Performing Test fstack-protector-strong
-- Performing Test fstack-protector-strong - Success
-- Performing Test Wimplicit-fallthrough
-- Performing Test Wimplicit-fallthrough - Failed

------------------------------------------------------------------------
-- Configuring solidity
------------------------------------------------------------------------
--                  CMake Version                            3.7.0
-- CMAKE_BUILD_TYPE Build type                               
-- TARGET_PLATFORM  Target platform                          Linux
--------------------------------------------------------------- features
-- PROFILING        Profiling support                        OFF
------------------------------------------------------------- components
-- TESTS            Build tests                              ON
------------------------------------------------------------------------

No SMT solver found (Z3 or CVC4). Optional SMT checking will not be available. Please install Z3 or CVC4 if it is desired.
-- Configuring done
-- Generating done
-- Build files have been written to: /root/solidity/build


/root/solidity/libevmasm/GasMeter.h:117:89:   required from here
/usr/include/c++/4.8.2/tuple:132:22: エラー: value-initialization of reference type ‘const boost::multiprecision::number<boost::multiprecision::backends::cpp_int_backend<256u, 256u, (boost::multiprecision::cpp_integer_type)0u, (boost::multiprecision::cpp_int_check_type)0u, void> >&’
       : _M_head_impl() { }
                      ^
make[2]: *** [libevmasm/CMakeFiles/evmasm.dir/Assembly.cpp.o] エラー 1
make[1]: *** [libevmasm/CMakeFiles/evmasm.dir/all] エラー 2
make: *** [all] エラー 2

```

--- a/libevmasm/GasMeter.h
+++ b/libevmasm/GasMeter.h
@@ -112,9 +112,10 @@ public:
 		static GasConsumption infinite() { return GasConsumption(0, true); }
 
 		GasConsumption& operator+=(GasConsumption const& _other);
-		bool operator<(GasConsumption const& _other) const { return this->tuple() < _other.tuple(); }
-
-		std::tuple<bool const&, u256 const&> tuple() const { return std::tie(isInfinite, value); }
+		bool operator<(GasConsumption const& _other) const
+		{
+			return std::make_pair(isInfinite, value) < std::make_pair(_other.isInfinite, _other.value);
+		}
 
 		u256 value;
 		bool isInfinite;

